src/Waives.Http/HttpRequestMessageTemplate.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Waives.Http
+{
+    public class HttpRequestMessageTemplate
+    {
+        public HttpRequestMessageTemplate(HttpMethod method, Uri requestUri)
+        {
+            Method = method ?? throw new ArgumentNullException(nameof(method));
+            RequestUri = requestUri ?? throw new ArgumentNullException(nameof(requestUri));
+            Headers = new List<KeyValuePair<string, string>>();
+        }
+
+        public HttpMethod Method { get; set; }
+        public Uri RequestUri { get; set; }
+        public HttpContent Content { get; set; }
+        public List<KeyValuePair<string, string>> Headers { get; }
```

Comment:
💪  There's a more specific type used for HTTP request headers, I think `HttpHeaderValueCollection`. This provides convenience members for adding well-known headers, and I think it would be better to program against this than a list of string pairs.

Reply:
Unfortunately it has no public `Add` method, so can't be used by us for these purposes I think.